providers/common/sql/src/airflow/providers/common/sql/hooks/sql.py

```diff
@@ -205,7 +205,7 @@ def placeholder(self) -> str:
         return self._placeholder
 
     @property
-    def insert_statement_format(self) -> str:
+    def insert_statement_format(self) -> str | None:
```

Comment:
Won't this always be a str though? If it's falsey, it goes looks at the connection extra, does a get and has a fallback, this returning a str.

MyPy might be happier with a `if self._insert_statement_format is None:`?

Reply:
Hmm yeah, it has a clear fall back defined, so I think mypy will be happy with that check too. Let me try

Reply:
Actually, problem seems to be in the typing in the usage. (__init__). 
It is defined as optional when it shouldn't be